pkg/controller/backup_controller.go

```diff
@@ -267,7 +267,6 @@ func (b *backupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	b.backupTracker.Add(request.Namespace, request.Name)
-	defer b.backupTracker.Delete(request.Namespace, request.Name)
```

Comment:
Previously, the behavior is like --- as long as the execution reaches here, and when it quits for whatever reason, the backupTracker will always be deleted.
Now, if we change like this, if someone insert some code between line 270 and 290 to return earlier, the backupTracker will not be deleted for that code. As a result, the backup will never be deletable for that case.

Reply:
One suggest here is that we still use the defer function, but inside it, we check the phase is not `WaitForAsyncOperation` then we delete the backupTracker.

Reply:
@Lyndon-Li That works too. I wanted to make sure we only closed it for terminal phases, but you're right -- putting that check in the defer func is cleaner. I'll update the PR.